integration/gethnetwork/geth_network.go

```diff
@@ -353,3 +356,39 @@ func waitForIPC(dataDir string) {
 		counter++
 	}
 }
+
+func EnsurePortsAreAvailable(startPort int, numberNodes int) error {
+	var unavailablePorts []int
+	mu := &sync.Mutex{}
+	wg := &sync.WaitGroup{}
+
+	for i := 0; i < numberNodes; i++ {
+		wg.Add(2)
+		go ensurePortAvailable(wg, mu, &unavailablePorts, startPort+i)              // commsPort
```

Comment:
@BedrockSquirrel Do we gain much by parallelising here? Seems to add complexity for little benefit AFAICT.

Reply:
Yeah agreed, I'll simplify. Felt like they were IO operations but actually how long does it take to listen and then immediately stop listening on a port. Was a useful exercise for me though, hadn't really dabbled in the black art of Go concurrency.